src/query/sql/src/planner/metadata.rs

```diff
@@ -420,7 +419,12 @@ impl Metadata {
                         field.name(),
                         display_tuple_field_name(inner_field_name)
                     );
-                    let inner_field = TableField::new(&inner_name, inner_field_type.clone());
+                    inner_column_id += inner_field_type.num_leaf_columns() as u32;
```

Comment:
The `inner_column_id` should be added by 1, and then if `num_leaf_columns` are greater than 1, then `num_leaf_columns` - 1 should be added after the inner field is generated, used to correctly generate the next `inner_column_id`.

Reply:
Thanks !